src/main/java/com/github/chhsiao90/nitmproxy/handler/protocol/tls/TlsFrontendHandler.java

```diff
@@ -88,6 +87,36 @@ private void configHttp2(ChannelHandlerContext ctx) {
     ctx.pipeline().replace(this, null, connectionContext.handler(Handler.HTTP2_FRONTEND));
   }
 
+  private class DetectSslHandler extends SslClientHelloHandler<Boolean> {
+
+    @Override
+    protected Future<Boolean> lookup(ChannelHandlerContext ctx, ByteBuf byteBuf) throws Exception {
+      boolean ssl = byteBuf != null;
+      LOGGER.debug("SSL detection with {}", ssl);
+      return ctx.executor().newSucceededFuture(ssl);
+    }
+
+    @Override
+    protected void onLookupComplete(ChannelHandlerContext ctx, Future<Boolean> future) throws Exception {
+      if (!future.isSuccess()) {
+        LOGGER.debug("SSL detection failed with {}", future.cause().getMessage());
+        ctx.close();
+      }
+      else {
+        if (!future.getNow()) {
+          connectionContext.tlsCtx().setPlain(true);
+          connectionContext.tlsCtx().protocolsPromise().setSuccess(Collections.singletonList(ApplicationProtocolNames.HTTP_1_1));
+          ctx.pipeline().remove(DetectSslHandler.class);
+          ctx.pipeline().remove(AlpnNegotiateHandler.class);
+          ctx.pipeline().remove(SniExtractorHandler.class);
+          configHttp1(ctx);
```

Comment:
I think the problem can be fixed by removing the handlers before removing the handlers.